bokeh/io.py

```diff
@@ -298,11 +303,25 @@ def show(obj, browser=None, new="tab", notebook_handle=False):
         an IPython/Jupyter notebook.
 
     '''
+    if isinstance(obj, Application):
+        return _show_notebook_app_with_state(obj, _state, app_path, notebook_url)
+
     if obj not in _state.document.roots:
         _state.document.add_root(obj)
     return _show_with_state(obj, _state, browser, new, notebook_handle=notebook_handle)
 
 
+def _show_notebook_app_with_state(app, _state, app_path, notebook_url):
+    from IPython.display import HTML, display
+    from tornado.ioloop import IOLoop
+    loop = IOLoop.current()
+    server = Server({'/': app}, io_loop=loop, port=0, host='*',
+                    allow_websocket_origin=[notebook_url])
```

Comment:
I just realized, `host='*'` is going to throw out a warning? We should find a way for that to not happen.